build/source/engine/vegNrgFlux.f90

```diff
@@ -1148,37 +991,26 @@ subroutine vegNrgFlux(&
-     if(err/=0)then; message=trim(message)//trim(cmessage); return; end if
-
-
-    ! assign scalar resistances for un-perturbed cases
-    else
+     if (err/=0) then; message=trim(message)//trim(cmessage); return; end if
+    else ! assign scalar resistances for un-perturbed cases
      trialLeafResistance   = scalarLeafResistance
      trialGroundResistance = scalarGroundResistance
      trialCanopyResistance = scalarCanopyResistance
-
-    end if  ! (re-computing resistances for perturbed cases)
-    !print*, 'trialLeafResistance = ', trialLeafResistance
-    !print*, 'trialGroundResistance = ', trialGroundResistance
-    !print*, 'trialCanopyResistance = ', trialCanopyResistance
-
+    end if  ! end re-computing resistances for perturbed cases
     ! compute the relative humidity in the top soil layer and the resistance at the ground surface
     ! NOTE: computations are based on start-of-step values, so only compute for the first flux call
-    if(firstFluxCall)then
-     ! (soil water evaporation factor [0-1])
-     soilEvapFactor = mLayerVolFracLiq(nSnow+1)/(theta_sat - theta_res)
-     ! (resistance from the soil [s m-1])
-     scalarSoilResistance = scalarGroundSnowFraction*1._rkind + (1._rkind - scalarGroundSnowFraction)*EXP(8.25_rkind - 4.225_rkind*soilEvapFactor)  ! Sellers (1992)
-     !scalarSoilResistance = scalarGroundSnowFraction*0._rkind + (1._rkind - scalarGroundSnowFraction)*exp(8.25_rkind - 6.0_rkind*soilEvapFactor)    ! Niu adjustment to decrease resitance for wet soil
-     ! (relative humidity in the soil pores [0-1])
-     if(mLayerMatricHead(1) > -1.e+6_rkind)then  ! avoid problems with numerical precision when soil is very dry
-      soilRelHumidity_noSnow = exp( (mLayerMatricHead(1)*gravity) / (groundTemp*R_wv) )
+    if (firstFluxCall) then
+     soilEvapFactor = mLayerVolFracLiq(nSnow+1)/(theta_sat - theta_res) ! soil water evaporation factor [0-1]
+     ! resistance from the soil [s m-1] -- Sellers (1992)
+     scalarSoilResistance = scalarGroundSnowFraction + (1._rkind - scalarGroundSnowFraction)*exp(8.25_rkind - 4.225_rkind*soilEvapFactor) ! removed factor of unity
+     ! relative humidity in the soil pores [0-1]
+     if (mLayerMatricHead(1) > -1.e+6_rkind) then  ! avoid problems with numerical precision when soil is very dry
+      soilRelHumidity_noSnow = exp((mLayerMatricHead(1)*gravity)/(groundTemp*R_wv))
      else
       soilRelHumidity_noSnow = 0._rkind
-     end if ! (if matric head is very low)
-     scalarSoilRelHumidity  = scalarGroundSnowFraction*1._rkind + (1._rkind - scalarGroundSnowFraction)*soilRelHumidity_noSnow
-     !print*, 'mLayerMatricHead(1), scalarSoilRelHumidity = ', mLayerMatricHead(1), scalarSoilRelHumidity
-    end if  ! (if the first flux call)
+     end if ! if matric head is very low
+     scalarSoilRelHumidity  = scalarGroundSnowFraction + (1._rkind - scalarGroundSnowFraction)*soilRelHumidity_noSnow ! removed factor of unity
```

Comment:
As above

Reply:
I have expanded the comment to give the original equation for context.